tests/end-to-end/external-solution-integration/pkg/testsuite/register_application_in_compass.go

```diff
@@ -34,10 +36,12 @@ type RegisterApplicationInCompassState interface {
 var _ step.Step = &RegisterApplicationInCompass{}
 
 // NewRegisterApplicationInCompass returns new RegisterApplicationInCompass
-func NewRegisterApplicationInCompass(name, apiURL string, applications acClient.ApplicationInterface, director *testkit.CompassDirectorClient, state RegisterApplicationInCompassState) *RegisterApplicationInCompass {
+func NewRegisterApplicationInCompass(name, apiURL string, applications acClient.ApplicationInterface,
+	director *testkit.CompassDirectorClient, state RegisterApplicationInCompassState, httpSources esClient.HTTPSourceInterface) *RegisterApplicationInCompass {
```

Comment:
It would be more logical to pass clients one after another in the constructor parameters. The current order `client, client, state, client` feels a bit strange to be.